docs/how-to/rocks/add-internal-user-in-rocks.rst

```diff
@@ -0,0 +1,45 @@
+.. _add_internal_user_in_rocks:
+
+How to add an internal user to a rock
+*************************************
+
+Rockcraft allows you to use ``run-user`` to specify the user you want to the
+rock's service(s) with. Without using that field in your rockcraft.yaml, the
+service(s) will be run by the root user. The downside, however, of using
+``run-user`` is that only supports one user called ``_daemon_`` with a UID/GID
+of 84792. For some daemon services that will be enough, but some other services
+require that the running user should have a specific username and/or UID.
+
+In order to add new internal user or group to a rock, three packages (or their
+respective slices) are needed:
+- ``base-files``: to create the base directories such as ``/etc`` and ``/bin``
```

Comment:
Why do these need to be staged? Is it insufficient to simply use `useradd` and `groupadd` to modify the relevant `/etc` files in the overlay?

Reply:
see comment below.